circuits/cpp/barretenberg/cpp/src/barretenberg/proof_system/relations/ecc_vm/ecc_set_relation.cpp

```diff
@@ -0,0 +1,395 @@
+#include "barretenberg/honk/flavor/ecc_vm.hpp"
+#include "barretenberg/honk/sumcheck/relation_definitions_fwd.hpp"
+#include "barretenberg/proof_system/relations/relation_parameters.hpp"
+#include "ecc_msm_relation.hpp"
+
+namespace proof_system::honk::sumcheck {
+
+template <typename FF>
+template <typename AccumulatorTypes>
+typename ECCVMSetRelationBase<FF>::template Accumulator<AccumulatorTypes> ECCVMSetRelationBase<
+    FF>::compute_permutation_numerator(const auto& extended_edges,
+                                       const RelationParameters<FF>& relation_params,
+                                       const size_t index)
+{
+    using Accumulator = typename std::tuple_element<0, typename AccumulatorTypes::Accumulators>::type;
+
+    const auto& precompute_round = get_view<FF, AccumulatorTypes>(extended_edges.precompute_round, index);
+    const auto precompute_round2 = precompute_round + precompute_round;
+    const auto precompute_round4 = precompute_round2 + precompute_round2;
+
+    const auto& gamma = relation_params.gamma;
+    const auto& beta = relation_params.beta;
+    const auto& beta_sqr = relation_params.beta_sqr;
+    const auto& beta_cube = relation_params.beta_cube;
+    const auto& precompute_pc = get_view<FF, AccumulatorTypes>(extended_edges.precompute_pc, index);
+    const auto& precompute_select = get_view<FF, AccumulatorTypes>(extended_edges.precompute_select, index);
+
+    /**
+     * @brief First term: tuple of (pc, round, wnaf_slice), computed when slicing scalar multipliers into slices,
+     *        as part of ECCVMWnafRelation.
+     *        If precompute_select = 1, tuple entry = (wnaf-slice + point-counter * beta + msm-round * beta_sqr).
+     *                       There are 4 tuple entires per row.
+     */
+    Accumulator numerator(1); // degree-0
+    {
+        const auto& s0 = get_view<FF, AccumulatorTypes>(extended_edges.precompute_s1hi, index);
+        const auto& s1 = get_view<FF, AccumulatorTypes>(extended_edges.precompute_s1lo, index);
+
+        auto wnaf_slice = s0 + s0;
+        wnaf_slice += wnaf_slice;
+        wnaf_slice += s1;
+
+        // todo can optimize
```

Comment:
Issues?

Reply:
What's the issue? Sorry could you clarify?

Reply:
Could you create issues for these todos?